server/core/src/actors/v1_read.rs

```diff
@@ -230,15 +232,21 @@ impl QueryServerReadV1 {
                 })?;
         }
 
+        // TODO: make the file rotation a separate function
+
         // pattern to find automatically generated backup files
-        let re = Regex::new(r"^backup-\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{1,9})?Z\.json$")
-            .map_err(|error| {
-                error!(
-                    "Failed to parse regexp for online backup files: {:?}",
-                    error
-                );
-                OperationError::InvalidState
-            })?;
+        let re = Regex::new(
+            &(r"^backup-\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d{1,9})?Z\.json".to_string()
+                + compression.suffix()
```

Comment:
It should find both compressed and uncompressed, where as won't this currently skip compressed if you change to uncompressed and vice versa?

Reply:
Yeah, I had considered that, will fix it